modules/sdk-coin-atom/src/atom.ts

```diff
@@ -226,6 +257,223 @@ export class Atom extends BaseCoin {
     };
   }
 
+  /**
+   * Builds a funds recovery transaction without BitGo
+   * @param {RecoveryOptions} params parameters needed to construct and
+   * (maybe) sign the transaction
+   *
+   * @returns {AtomTx} the serialized transaction hex string and index
+   * of the address being swept
+   */
+  async recover(params: RecoveryOptions): Promise<AtomTx> {
+    let publicKey: string;
+    // Step 1: Check if params contains the required parameters
+    if (!params.bitgoKey) {
+      throw new Error('missing bitgoKey');
+    }
+
+    if (!params.recoveryDestination || !this.isValidAddress(params.recoveryDestination)) {
+      throw new Error('invalid recoveryDestination');
+    }
+
+    // Step 2: Fetch the starting scan index if its present (else 0)
+    let startIdx = params.startingScanIndex;
+    if (_.isUndefined(startIdx)) {
+      startIdx = 0;
+    } else if (!isInteger(startIdx) || startIdx < 0) {
+      throw new Error('Invalid starting index to scan for addresses');
+    }
+
+    // Step 3: Fetch the number of iterations from params if its present (else 20)
+    let numIteration = params.scan;
+    if (_.isUndefined(numIteration)) {
+      // magic number ? Why 20 ?
+      numIteration = 20;
+    } else if (!isInteger(numIteration) || numIteration <= 0) {
+      throw new Error('Invalid scanning factor');
+    }
+
+    // Step 4: Fetch the bitgo key from params
+    const bitgoKey = params.bitgoKey.replace(/\s/g, '');
+
+    // Step 5: Instantiate the ECDSA signer
+    const MPC = new Ecdsa();
+
+    // Step 6: For each account from the starting index till numIterations + starting index, fetch the build tx details
+    for (let i = startIdx; i < numIteration + startIdx; i++) {
+      const currPath = `m/${i}`;
+      publicKey = MPC.deriveUnhardened(bitgoKey, currPath).slice(0, 66);
+      const senderAddress = this.getAddressFromPublicKey(publicKey);
+      const balance = await this.getAccountBalance(senderAddress);
+
+      if (Number(balance) <= 0) {
```

Comment:
Will the `balance` always be within the number range?

Reply:
Changed to `BigNumber` to avoid problems